### PR TITLE
Fixed wrong return from getScaledAssetPath()

### LINKED ISSUE #3980: new asset system doesn't work in windows.
I tried to present just an image in windows 10.

I added an image in my project directory -> ./image/base.png

With the render function below:

    <View style={styles.container}>
        <Image
        style={{width:80, height:80}}
        source={require('./image/base.png')}
		/>
    </View>

I can't see any image with it.

I tried in OS X, it works well.

### PR DESCRIPTION
**Pull request by WoLfulus**
Fixes #3980

### CODE REVIEW

**Comment by nicklockwood** [comment]
It seems like this should be fixed inside the getBasePath() function, rather than at this call site. @frantic?

**Comment by frantic** [comment]
@WoLfulus thanks for the PR! @nicklockwood is right, `getBasePath` is a much better place for this. However, seems like the issue is much deeper in the stack, because the only way the first character of path can be `\` is when `asset.httpServerLocation` starts with `\`, which doesn't make much sense because http location should only include `/`.

To help us debug, could you please add `console.log` before `assetToImageSource` call and post the output here?

**Comment by WoLfulus** [comment]
@frantic I was expecting to see the output on the packager window, but nothing shows up. Where and when exactly the URLs are made?

**Comment by WoLfulus** [comment]
@frantic I got a log.

See at : http://pastebin.com/yakrrCnL

I didn't know exactly what you meant by "before" assetToImageSource, but this is how I made the log.

	function assetToImageSource(asset): ResolvedAssetSource {
		var devServerURL = getDevServerURL();
		var ret = {
			__packager_asset: true,
			width: asset.width,
			height: asset.height,
			uri: devServerURL ? getPathOnDevserver(devServerURL, asset) : getPathInArchive(asset),
			scale: pickScale(asset.scales, PixelRatio.get()),
		};
		console.log(JSON.stringify({
			'devServerURL': devServerURL,
			'asset': asset,
			'return': ret
		}, null, 4));
		return ret;
	}

Hope it helps somehow.

Also, images doesn't load if debugger is attached and I have no idea why. Maybe a different bug? Because it doesn't seems to make a request.

**Comment by frantic** [comment]
@WoLfulus thank you, it helps! The root of the problem is here https://github.com/facebook/react-native/blob/master/packager/react-packager/src/Bundler/index.js#L366 – the packager uses `path.join` to format `httpServerLocation`, and `path.join` will use `\` on Windows. If you are willing to help, please update your pull request to fix the root cause in `Bundler/index.js`. From quick Googling you can either regex-replace `\` with `/` or use [`url.resolve`](https://nodejs.org/api/url.html#url_url_resolve_from_to)

I'll look into images and debugger, must be a different problem

**Comment by WoLfulus** [comment]
@frantic I dont think thats the only problem. Basically I made the patch because on Windows the image url was being made with a double "/". I'll leave both changes. Is that alright?

File: packager/react-packager/src/Bundler/index.js [hunk 1]
```diff
@@ -303,7 +303,7 @@ class Bundler {
       const img = {
         __packager_asset: true,
         fileSystemLocation: path.dirname(module.path),
-        httpServerLocation: path.join('/assets', path.dirname(relPath)),
+        httpServerLocation: path.join('/assets', path.dirname(relPath)).replace(/\\/g, "/"),
```
Comment (frantic) [inline packager/react-packager/src/Bundler/index.js]: nit: prefer `'`

**Comment by frantic** [comment]
> on Windows the image url was being made with a double "/"

We need to find the root cause for this. Maybe you could provide another pastebin after your patch has been applied? `httpServerLocation` should look like `/assets/a/b/c.png`, if it doesn't, we need to fix packager.

**Comment by mkonicek** [comment]
Closing as the linked issue is closed and based on the comment above. Let me know if I'm mistaken :)

> Great! It does work!

**Pull request closed without merging**